app/models/manageiq/providers/kubernetes/container_manager/scanning/job.rb

```diff
@@ -46,6 +46,11 @@ def start
     image = target_entity
     return queue_signal(:abort_job, "no image found", "error") unless image
     return queue_signal(:abort_job, "cannot analyze non docker images", "error") unless image.docker_id
+    return queue_signal(
+      :abort_job,
+      "smartproxy role missing",
+      "error"
+    ) unless MiqServer.my_server.has_active_role?("smartproxy")
```

Comment:
@jrafanie  In order for this job to succeed it must have the smartproxy role.
 We want to enforce the smartstate as well. [Here](https://github.com/ManageIQ/manageiq/blob/master/app/models/job_proxy_dispatcher.rb#L134) is the only relevant code I could find that deals with roles for this job. I can see it in my environment being picked up by a server with out smartproxy which makes sense since because it is only placed with smartstate.

- How can I make sure a job is only picked up by a server with smartproxy+smartstate?
- Is looks to me like [vm scans](https://github.com/ManageIQ/manageiq/blob/master/app/models/job_proxy_dispatcher.rb#L170) only use smartproxy in case they need a proxy. Is that a good fix here too?